### ISSUE TITLE
Issue with heatmap notify event after A21.1

### ISSUE THREAD

**Issue by FilUnderscore** [FilUnderscore/ImprovedHordes#19]
Looks like a NRE in the notify event method.

```
2023-08-10T21:09:55 363.237 ERR [XUi] Error while updating window group 'workstation_campfire':2023-08-10T21:09:55 363.237 EXC Object reference not set to an instance of an object
at ImprovedHordes.Core.World.Event.WorldEventReporter+AIDirectorChunkEventComponent_NotifyEvent_Patch.Postfix (AIDirectorChunkEvent _chunkEvent) [0x00045] in <f3df1dba03e14536a032e6cdeaed5c4b>:0
at (wrapper dynamic-method) AIDirectorChunkEventComponent.DMD<AIDirectorChunkEventComponent::NotifyEvent>(AIDirectorChunkEventComponent,AIDirectorChunkEvent)
at AIDirector.NotifyActivity (EnumAIDirectorChunkEvent type, Vector3i position, System.Single value, System.Single _duration) [0x00039] in <d7dcfa4979674f179c13273f8b30b551>:0
at TileEntity.emitHeatMapEvent (World world, EnumAIDirectorChunkEvent eventType) [0x00046] in <d7dcfa4979674f179c13273f8b30b551>:0
at TileEntityWorkstation.HandleFuel (World _world, System.Single _timePassed) [0x0000a] in <d7dcfa4979674f179c13273f8b30b551>:0
at (wrapper dynamic-method) TileEntityWorkstation.DMD<TileEntityWorkstation::UpdateTick>(TileEntityWorkstation,World)
at XUiC_WorkstationWindowGroup.Update (System.Single _dt) [0x0006c] in <d7dcfa4979674f179c13273f8b30b551>:0
at XUi.OnUpdateDeltaTime (System.Single updateDeltaTime) [0x00131] in <d7dcfa4979674f179c13273f8b30b551>:0
UnityEngine.StackTraceUtility:ExtractStringFromException(Object)
Log:Exception(Exception)
XUi:OnUpdateDeltaTime(Single)
XUiUpdater:Update()
XUiUpdateHelper:LateUpdate() 
```

**Comment by FilUnderscore** [comment]
Might also be linked to unexpected behavior of heat generating tile entities continuing to burn only when the player is inside the menu.

**Comment by FilUnderscore** [comment]
Fix included in commit 226f7fb.

**Comment by FilUnderscore** [comment]
Fixed in latest release 2.0.0-beta.5.

Changelog note:
> Fixed: NotifyEvent patch causing NullReferenceException leading to bugs with heatmap emitting tile entities.
